### PR TITLE
fixed issue with allocation_source calculation

### PR DESCRIPTION
**Pull request by amit4111989**

### CODE REVIEW

File: service/allocation_logic.py [hunk 1]
```diff
@@ -97,8 +97,8 @@ def map_events_to_histories(filtered_instance_histories, event_instance_dict):
                 out_dic.setdefault(inst_history[-1], []).append(info)
     return out_dic
 
-def get_allocation_source_name_from_event(username, report_start_date):
-    events = EventTable.objects.filter(Q(timestamp__lt=report_start_date) & Q(name__exact="instance_allocation_source_changed") & Q(payload__username__exact=username)).order_by('timestamp')
+def get_allocation_source_name_from_event(username, report_start_date, instance_id):
+    events = EventTable.objects.filter(Q(timestamp__lt=report_start_date) & Q(name__exact="instance_allocation_source_changed") & Q(payload__username__exact=username) & Q(payload__instance_id__exact=instance_id)).order_by('timestamp')
```
Comment (julianpistorius) [inline service/allocation_logic.py]: TODO: Modify this query to order descending, and limit to 1.

**Comment by julianpistorius** [comment]
Looks good, other than that little optimization.

**Comment by amit4111989** [comment]
Don't merge now. May need testing

**Comment by steve-gregory** [comment]
We have abandoned the branch you are attempting to merge into.

**Pull request closed without merging**